ilmoituslomake/opening_times/views.py

```diff
@@ -0,0 +1,138 @@
+                return update_response
+        else:
+            # If kaupunkialusta:id cannot be found in hauki, search for the pure hauki_id.
+            hauki_id_response = requests.get(HAUKI_API_URL + hauki_id + "/")
+            # Find the notification from the kaupunkialusta db.
+            notification = get_object_or_404(Notification, pk=id)
+            # If the hauki_id can be found from hauki and the notification is moderated,
+            # update the origin of the hauki resource, otherwise create a new hauki resource.
+            if (
+                hauki_id_response.status_code == 200
+                and notification.moderated_notification_id > 0
+            ):
+                update_origin(id, notification.moderated_notification_id)
+                # Update hsa_resource for the url creation.
+                hsa_resource = "kaupunkialusta:" + str(
+                    notification.moderated_notification_id
+                )
+            elif notification.moderated_notification_id > 0:
+                # Create data at v1_resource_create
+                origins = {
+                    "data_source": {
+                        "id": "kaupunkialusta",
+                    },
+                    "origin_id": notification.moderated_notification_id,
+                }
+                create_response = create_hauki_resource(
+                    name,
+                    description,
+                    address,
+                    resource_type,
+                    origins,
+                    is_public,
+                    timezone,
+                )
+                hsa_resource = "kaupunkialusta:" + str(
+                    notification.moderated_notification_id
+                )
+                if create_response.status_code != 201:
+                    return create_response
+
+        # Now time used for link expiration and creation time
+        now = datetime.utcnow().replace(microsecond=0)
+
+        # Construct the url
+        url_data = {
+            "hsa_source": "kaupunkialusta",
+            "hsa_username": request.user.email,
+            "hsa_created_at": now.isoformat() + "Z",
+            "hsa_valid_until": (now + timedelta(hours=1)).isoformat() + "Z",
+            "hsa_organization": "tprek:0c71aa86-f76c-466b-b6f3-81143bd9eecc",
+            "hsa_resource": hsa_resource,
+            "hsa_has_organization_rights": "false",
+        }
+        url = create_url(url_data)
+
+        return Response(url, status=status.HTTP_200_OK)
+
+
+class GetTimes(RetrieveAPIView):
+    """
+    Endpoint for getting opening hours of a hauki resource.
+    """
+
+    queryset = ""
+
+    def get(self, request, id=None, *args, **kwargs):
+        if not request.user:
+            return Response(None, status=status.HTTP_403_FORBIDDEN)
+        start_date = self.request.query_params.get("start_date", None)
+        end_date = self.request.query_params.get("end_date", None)
+        response = requests.get(
```

Comment:
Add some small but reasonable timeout value here. We don't want to wait too long here and handle the error case.